Error synching table without primary key

Hello,

I have this exception when trying to sync a table that there is no primary key:

```java
2025-03-11 11:31:24,723 ERROR [io.deb.emb.asy.AsyncEmbeddedEngine] (pool-7-thread-1) Engine has failed with : java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "com.fasterxml.jackson.databind.JsonNode.get(String)" because the return value of "io.debezium.server.bigquery.BaseRecordConverter.keySchema()" is null
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
        at io.debezium.embedded.async.AsyncEmbeddedEngine.runTasksPolling(AsyncEmbeddedEngine.java:489)
        at io.debezium.embedded.async.AsyncEmbeddedEngine.run(AsyncEmbeddedEngine.java:220)
        at io.debezium.server.DebeziumServer.lambda$start$1(DebeziumServer.java:180)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.NullPointerException: Cannot invoke "com.fasterxml.jackson.databind.JsonNode.get(String)" because the return value of "io.debezium.server.bigquery.BaseRecordConverter.keySchema()" is null
        at io.debezium.server.bigquery.BaseRecordConverter.keyFields(BaseRecordConverter.java:183)
        at io.debezium.server.bigquery.BaseRecordConverter.tableConstraints(BaseRecordConverter.java:220)
        at io.debezium.server.bigquery.StreamBigqueryChangeConsumer.getTable(StreamBigqueryChangeConsumer.java:260)
        at io.debezium.server.bigquery.StreamBigqueryChangeConsumer.uploadDestination(StreamBigqueryChangeConsumer.java:148)
        at io.debezium.server.bigquery.BaseChangeConsumer.handleBatch(BaseChangeConsumer.java:122)
        at io.debezium.embedded.async.ParallelSmtAndConvertBatchProcessor.processRecords(ParallelSmtAndConvertBatchProcessor.java:56)
        at io.debezium.embedded.async.AsyncEmbeddedEngine$PollRecords.doCall(AsyncEmbeddedEngine.java:1191)
        at io.debezium.embedded.async.AsyncEmbeddedEngine$PollRecords.doCall(AsyncEmbeddedEngine.java:1172)
        at io.debezium.embedded.async.RetryingCallable.call(RetryingCallable.java:47)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        ... 3 more

2025-03-11 11:31:24,725 INFO  [io.deb.emb.asy.AsyncEmbeddedEngine] (pool-7-thread-1) Engine state has changed from 'POLLING_TASKS' to 'STOPPING'
```

The table DDL in question:

```sql
create table api_key_clusters
(
    key_id  varchar(255) not null
        constraint fkc7oyefdt1tj7gcc2bdtkuqudj
            references api_key,
    cluster varchar(255)
);
```

Any idea how to workaround it ?

Thank you

--- a/debezium-server-bigquery-sinks/src/main/java/io/debezium/server/bigquery/BaseRecordConverter.java
+++ b/debezium-server-bigquery-sinks/src/main/java/io/debezium/server/bigquery/BaseRecordConverter.java
@@ -178,10 +178,12 @@ public abstract class BaseRecordConverter implements RecordConverter {
   }
 
   protected ArrayList<String> keyFields() {
-
     ArrayList<String> keyFields = new ArrayList<>();
-    for (JsonNode jsonSchemaFieldNode : this.keySchema().get("fields")) {
-      keyFields.add(jsonSchemaFieldNode.get("field").textValue());
+
+    if (this.keySchema() != null && this.keySchema().has("field")) {
+      for (JsonNode jsonSchemaFieldNode : this.keySchema().get("fields")) {
+        keyFields.add(jsonSchemaFieldNode.get("field").textValue());
+      }
     }
 
     return keyFields;

--- a/debezium-server-bigquery-sinks/src/test/java/io/debezium/server/bigquery/StreamBigqueryChangeConsumerMysqlTest.java
+++ b/debezium-server-bigquery-sinks/src/test/java/io/debezium/server/bigquery/StreamBigqueryChangeConsumerMysqlTest.java
@@ -47,8 +47,7 @@ public class StreamBigqueryChangeConsumerMysqlTest extends BaseBigqueryTest {
         " c_id2 INTEGER ," +
         " c_data TEXT," +
         " c_text TEXT," +
-        " c_varchar VARCHAR(1666) ," +
-        " PRIMARY KEY (c_id, c_id2)" +
+        " c_varchar VARCHAR(1666) " +
         " );";
     SourceMysqlDB.runSQL(createTable);
     String sqlInsert =
@@ -80,6 +79,7 @@ public class StreamBigqueryChangeConsumerMysqlTest extends BaseBigqueryTest {
     public Map<String, String> getConfigOverrides() {
       Map<String, String> config = new HashMap<>();
       config.put("debezium.sink.type", "bigquerystream");
+      config.put("debezium.sink.bigquerystream.upsert", "false");
       config.put("debezium.sink.bigquerystream.allow-field-addition", "true");
       config.put("debezium.source.table.include.list", "inventory.test_table");
       return config;

